c++11/src/propagation.cpp

```diff
@@ -37,9 +37,66 @@ static uint64_t hex_to_uint64(const std::string& s) {
 }
 
 //------------------------------------------------------------------------------
+// write_uint64
+//------------------------------------------------------------------------------
+static void write_uint64(std::ostream& out, uint64_t u) {
+  out.write(reinterpret_cast<char*>(&u), sizeof(u));
+}
+
+//------------------------------------------------------------------------------
+// read_uint64
+//------------------------------------------------------------------------------
+static uint64_t read_uint64(std::istream& in) {
+  uint64_t u = 0;
+  in.read(reinterpret_cast<char*>(&u), sizeof(u));
+  if (!in.good()) return 0;
+  return u;
+}
+
+//------------------------------------------------------------------------------
+// write_string
+//------------------------------------------------------------------------------
+static void write_string(std::ostream& out, const std::string& s) {
+  write_uint64(out, s.size());
+  out.write(s.data(), s.size());
+}
+
+//------------------------------------------------------------------------------
+// read_string
+//------------------------------------------------------------------------------
+static void read_string(std::istream& in, std::string& s) {
+  auto length = read_uint64(in);
+  s.resize(length);
+  in.read(&s.front(), length);
+}
+
+//------------------------------------------------------------------------------
 // inject_span_context
 //------------------------------------------------------------------------------
 Expected<void> inject_span_context(
+    std::ostream& carrier, uint64_t trace_id, uint64_t span_id,
```

Comment:
I see that later you replaced this code with https://github.com/lightstep/lightstep-tracer-common/blob/master/lightstep_carrier.proto#L6, but this made me think of asking you to hand-code the binary context encoding here, because it's a relatively simple subset of the protobuf encoding and pretty easy to write. I'll hold off on asking for that at this time. :)